### PR TITLE
Adds bls12 precompile to pallet_evm

### PR DESCRIPTION
**Pull request by drewstone**

### CODE REVIEW

**Review by sorpaas — COMMENTED** [review]
@drewstone Would you mind to move the `eth_pairings` dependency under a feature flag? I remember the implementation is quite big (also compile time).

The PR itself LGTM and I think it works. But practically, a production blockchain will never want to include those precompiles like this. Wasm bls12 is actually quite slow, from what I can recall. A production chain would need to use host functions to implement those calls.

File: frame/evm/Cargo.toml [hunk 1]
```diff
@@ -28,6 +28,7 @@ evm = { version = "0.17", default-features = false }
 sha3 = { version = "0.8", default-features = false }
 impl-trait-for-tuples = "0.1"
 ripemd160 = { version = "0.9", default-features = false }
+eth_pairings = { git = "https://github.com/drewstone/eip1962.git", default-features = false, features = ["eip_2537"] }
```
Comment (NikVolf) [inline frame/evm/Cargo.toml:31]: This should be published on crates.io

**Comment by burdges** [comment]
Anyone benchmark the pairing in wasm?  

As an aside, I think ethereum exposes only a simple pairing call, not the miller loop, which makes BLS signatures almost twice as expensive, and makes Groth16 cost like tree times more.  It's still maybe 300 times slower than ed25519, even in native though.

**Comment by drewstone** [comment]
Sounds good with the features, so far I don't think I implemented the `no_std` correctly as I'm getting the usual error. Will keep working on this.

**Comment by drewstone** [comment]
I didn't port the `no_std` correctly and haven't grok'd the process of doing so in that repo.. gonna close this for now to avoid wasting your time.

**Pull request closed without merging**